openedx/core/djangoapps/notifications/tests/test_tasks.py

```diff
@@ -291,10 +296,11 @@ def _create_users(self, num_of_users):
         return users
 
     @override_waffle_flag(ENABLE_NOTIFICATIONS, active=True)
+    @override_waffle_flag(ENABLE_PUSH_NOTIFICATIONS, active=True)
     @ddt.data(
-        (settings.NOTIFICATION_CREATION_BATCH_SIZE, 10, 4),
-        (settings.NOTIFICATION_CREATION_BATCH_SIZE + 10, 12, 7),
-        (settings.NOTIFICATION_CREATION_BATCH_SIZE - 10, 10, 4),
+        (settings.NOTIFICATION_CREATION_BATCH_SIZE, 10, 7),
+        (settings.NOTIFICATION_CREATION_BATCH_SIZE + 10, 12, 10),
```

Comment:
- I think query count should be +2 not +3. One query for fetching waffle flag and second to get list of emails.
- Purpose of this test case is to make sure that `NOTIFICATION_CREATION_BATCH_SIZE` is optimized. It is better to keep changes in this test case to minimum. i.e. we can set push waffle flag to False and see that if it only increases 1 query count. If change is more than +1, we need to update batch size accordingly

Reply:
Query count is +3. One for waffle flag, second one for all receiving users and third one is for sender.